### PR TITLE
process-agent: Add metrics on the # of opened connections (tcp, udp) per process / container

### PR DESCRIPTION
**Pull request by sfluor**
### What does this PR do?

Adds a metric to process / container payloads to count the number of TCP connections / UDP connections still opened.

Note that for now the # might not be accurate for TCP connections because of some tcp_close events we are missing and our aggressive internal expiry logic, [this could however fix the issue](https://github.com/DataDog/datadog-agent/pull/3728).

For UDP connections, they will be considered opened if a message was sent less than 30seconds ago (this is how the UDP expiry logic works currently).

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: pkg/process/checks/container.go [hunk 1]
```diff
@@ -156,6 +158,10 @@ func fmtContainers(ctrList []*containers.Container, lastRates map[string]util.Co
 			Addresses:   convertAddressList(ctr),
 			Started:     ctr.StartedAt,
 			Tags:        tags,
+			Conns: &model.ConnsStat{
+				OpenedTcp: connsCounts.TCP,
```
Comment (shang-wang) [inline pkg/process/checks/container.go]: Feel free not to do this, but I somehow like them to be direct fields on container struct, so I don't have to do `container.Conns.OpenedTcp`.

Comment (shang-wang) [reply pkg/process/checks/container.go]: Another reason is that protobuf would make the nested field a reference not an object, so it's not as GC friendly.

File: pkg/process/checks/net.go [hunk 2]
```diff
@@ -194,3 +230,25 @@ func getCtrIDsByPIDs(pids []int32) map[int32]string {
 	}
 	return Process.filterCtrIDsByPIDs(pids)
 }
+
+// countConnectionsPerPID counts the number of connections by type by PID
+func countConnectionsPerPID(conns []*model.Connection) map[int32]ConnectionsCounts {
+	res := map[int32]ConnectionsCounts{}
```
Comment (shang-wang) [inline pkg/process/checks/net.go]: Does this function need to use lock?

Comment (sfluor) [reply pkg/process/checks/net.go]: It's not linked to the check so It should be fine 👍

File: pkg/process/checks/container.go (hunk 1 above)
Comment (sfluor) [reply pkg/process/checks/container.go]: This was to avoid repetition on all the different objects (`process_rt`, `process`, `container_rt`, `container`)

**Pull request closed without merging**